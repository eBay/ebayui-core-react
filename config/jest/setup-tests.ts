import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
