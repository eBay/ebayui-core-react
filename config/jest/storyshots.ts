import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
