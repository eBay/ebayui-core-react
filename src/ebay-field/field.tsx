import React, { FC, ReactNode } from 'react'
import classNames from 'classnames'
import { FieldLayoutType } from './types'

type FieldProps = {
    className?: string;
    layout?: FieldLayoutType;
    children?: ReactNode;
};

const Field: FC<FieldProps> = ({
    className,
    layout = 'inline',
    children
}) => {
    const WrapperElement = layout === 'block' ? 'div' : 'span'
    return (
        <WrapperElement className={classNames('field', className)}>
            {children}
        </WrapperElement>
    )
}

export default Field
