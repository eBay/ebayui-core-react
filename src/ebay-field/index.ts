export { default as EbayField } from './field'
export { default as EbayLabel, Props as EbayLabelProps } from './label'
export { default as EbayFieldDescription } from './description'
export { FieldDescriptionType, FieldDescriptionPosition, FieldLayoutType, LabelPosition } from './types'
