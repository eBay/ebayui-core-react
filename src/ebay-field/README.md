# ebay-field

## Demo
[Storybook](https://opensource.ebay.com/ebayui-core-react/main/?path=/story/ebay-field--default-inline)

## Import JS
```jsx harmony
import { EbayField, EbayLabel, fieldLayoutType, EbayFieldDescription, fieldDescriptionType, fieldDescriptionPosition } from '@ebay/ui-core-react/ebay-field';
```
## Import following styles from SKIN
```jsx harmony
import '@ebay/skin/field';
```
## Import styles using SCSS/CSS
```jsx harmony
import '@ebay/skin/field.css'
```
## Usage
```
yarn add @ebay/ui-core-react
```
```jsx harmony
 <EbayField layout={fieldLayoutType.BLOCK}>
    <EbayLabel stacked htmlFor="field1">Label 1</EbayLabel>
    <EbayTextbox placeholder="placeholder text" id="field1" />
    <EbayFieldDescription
        type={fieldDescriptionType.CONFIRMATION}
        position={fieldDescriptionPosition.BELOW_FIELD}>
            Some description Text
    </EbayFieldDescription>
</EbayField>
```

## EbayField Attributes

Name | Type | Stateful | Required | Description
--- | --- | --- | --- | ---
`layout` | String | No | No | `fieldLayoutType.BLOCK`, `fieldLayoutType.inline` (default)

## EbayLabel Attributes

Name | Type | Stateful | Required | Description
--- | --- | --- | --- | ---
`stacked` | Boolean | No | No | display label above the field if true
`required` | Boolean | No | No | indicates the field is required if true
`position` | String | No | No | `start` (default) or `end` position towards the input

## EbayFieldDescription Attributes

Name | Type | Stateful | Required | Description
--- | --- | --- | --- | ---
`type` | String | No | No | `fieldDescriptionType.CONFIRMATION`, `fieldDescriptionType.DEFAULT`(Default), `fieldDescriptionType.ATTENTION`, `fieldDescriptionType.INFORMATION`
`position` | String | No | No | `fieldDescriptionPosition.BELOW_FIELD` (Default), `fieldDescriptionPosition.INLINE`
