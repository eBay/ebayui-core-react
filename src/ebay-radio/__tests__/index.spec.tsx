import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { initStoryshots } from '../../../config/jest/storyshots';
import { EbayRadio } from '../index';
import { EbayLabel } from '../../ebay-field'

initStoryshots({
    config: ({ configure }) =>
        configure(() => {require('./index.stories')}, module),
    storyNameRegex: /^((?!.*?controlled).)*$/i // all stories except the Controlled ones
})

var anySyntheticEvent = expect.objectContaining({ type: null });

describe('<EbayRadio>', () => {
    describe('on radio-button click', () => {
        it('should fire onChange event', () => {
            const value = 'test'
            const spy = jest.fn()
            render(<EbayRadio value={value} onChange={spy} id="id">
                <EbayLabel>radio</EbayLabel>
            </EbayRadio>)
            fireEvent.click(screen.getByLabelText('radio'))

            expect(spy).toBeCalledWith(anySyntheticEvent, { value })
        })
        it('should fire onFocus event', () => {
            const value = 'test'
            const spy = jest.fn()
            render(<EbayRadio value={value} onFocus={spy} id="id">
                <EbayLabel>radio</EbayLabel>
            </EbayRadio>)
            fireEvent.focus(screen.getByLabelText('radio'))

            expect(spy).toBeCalledWith(anySyntheticEvent, { value })
        })
    })
    describe('on radio-button key down', () => {
        it('should fire onKeyDown event', () => {
            const value = 'test'
            const spy = jest.fn()
            render(<EbayRadio value={value} onKeyDown={spy} id="id">
                <EbayLabel>radio</EbayLabel>
            </EbayRadio>)
            fireEvent.keyDown(screen.getByLabelText('radio'))

            expect(spy).toBeCalledWith(anySyntheticEvent, { value })
        })
    })
})
