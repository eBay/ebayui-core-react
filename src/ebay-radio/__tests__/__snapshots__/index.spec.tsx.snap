// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-radio Default 1`] = `
<DocumentFragment>
  <p>
    <span
      class="radio"
      style="align-items: center;"
    >
      <input
        class="radio__control"
        id="radio-1"
        type="radio"
        value="123"
      />
      <span
        class="radio__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="radio__checked icon icon--radio-checked"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-radio-checked"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="radio__unchecked icon icon--radio-unchecked"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-radio-unchecked"
          />
        </svg>
      </span>
    </span>
    <label
      class="field__label field__label--end"
      for="radio-1"
    >
      Default
    </label>
  </p>
  <p>
    <span
      class="radio radio--large"
      style="align-items: center;"
    >
      <input
        class="radio__control"
        id="radio-11"
        type="radio"
        value="123"
      />
      <span
        class="radio__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="radio__checked icon icon--radio-checked-large"
          focusable="false"
          height="64px"
          width="64px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-radio-checked-large"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="radio__unchecked icon icon--radio-unchecked-large"
          focusable="false"
          height="64px"
          width="64px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-radio-unchecked-large"
          />
        </svg>
      </span>
    </span>
    <label
      class="field__label field__label--end"
      for="radio-11"
    >
      Large
    </label>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-radio Disabled radio-button 1`] = `
<DocumentFragment>
  <span
    class="radio"
    style="align-items: center;"
  >
    <input
      class="radio__control"
      disabled=""
      id="radio-20"
      type="radio"
    />
    <span
      class="radio__icon"
      hidden=""
    >
      <svg
        aria-hidden="true"
        class="radio__checked icon icon--radio-checked"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-radio-checked"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="radio__unchecked icon icon--radio-unchecked"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-radio-unchecked"
        />
      </svg>
    </span>
  </span>
  <label
    class="field__label field__label--end"
    for="radio-20"
  >
    Disabled
  </label>
</DocumentFragment>
`;

exports[`Storyshots ebay-radio Grouped radio-buttons 1`] = `
<DocumentFragment>
  <fieldset>
    <legend>
      Choose an Option
    </legend>
    <span
      class="field"
    >
      <span
        class="radio"
        style="align-items: center;"
      >
        <input
          checked=""
          class="radio__control"
          id="group-radio-1"
          name="radio-group"
          type="radio"
          value="1"
        />
        <span
          class="radio__icon"
          hidden=""
        >
          <svg
            aria-hidden="true"
            class="radio__checked icon icon--radio-checked"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-radio-checked"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="radio__unchecked icon icon--radio-unchecked"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-radio-unchecked"
            />
          </svg>
        </span>
      </span>
      <label
        class="field__label field__label--end"
        for="group-radio-1"
      >
        Option 1
      </label>
    </span>
    <span
      class="field"
    >
      <span
        class="radio"
        style="align-items: center;"
      >
        <input
          class="radio__control"
          id="group-radio-2"
          name="radio-group"
          type="radio"
          value="2"
        />
        <span
          class="radio__icon"
          hidden=""
        >
          <svg
            aria-hidden="true"
            class="radio__checked icon icon--radio-checked"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-radio-checked"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="radio__unchecked icon icon--radio-unchecked"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-radio-unchecked"
            />
          </svg>
        </span>
      </span>
      <label
        class="field__label field__label--end"
        for="group-radio-2"
      >
        Option 2
      </label>
    </span>
    <span
      class="field"
    >
      <span
        class="radio"
        style="align-items: center;"
      >
        <input
          class="radio__control"
          id="group-radio-3"
          name="radio-group"
          type="radio"
          value="3"
        />
        <span
          class="radio__icon"
          hidden=""
        >
          <svg
            aria-hidden="true"
            class="radio__checked icon icon--radio-checked"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-radio-checked"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="radio__unchecked icon icon--radio-unchecked"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-radio-unchecked"
            />
          </svg>
        </span>
      </span>
      <label
        class="field__label field__label--end"
        for="group-radio-3"
      >
        Option 3
      </label>
    </span>
  </fieldset>
</DocumentFragment>
`;

exports[`Storyshots ebay-radio Selected radio-button 1`] = `
<DocumentFragment>
  <span
    class="radio"
    style="align-items: center;"
  >
    <input
      checked=""
      class="radio__control"
      id="radio-2"
      type="radio"
    />
    <span
      class="radio__icon"
      hidden=""
    >
      <svg
        aria-hidden="true"
        class="radio__checked icon icon--radio-checked"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-radio-checked"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="radio__unchecked icon icon--radio-unchecked"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-radio-unchecked"
        />
      </svg>
    </span>
  </span>
  <label
    class="field__label field__label--end"
    for="radio-2"
  >
    Selected
  </label>
</DocumentFragment>
`;

exports[`Storyshots ebay-radio Styled radio-button 1`] = `
<DocumentFragment>
  <style>
    
            .custom ~ label,
            .custom .radio__icon svg { color: green !important }
            
  </style>
  <span
    class="radio custom"
    style="align-items: center;"
  >
    <input
      aria-label="custom color radio example"
      class="radio__control"
      id="radio-30"
      type="radio"
    />
    <span
      class="radio__icon"
      hidden=""
    >
      <svg
        aria-hidden="true"
        class="radio__checked icon icon--radio-checked"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-radio-checked"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="radio__unchecked icon icon--radio-unchecked"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-radio-unchecked"
        />
      </svg>
    </span>
  </span>
  <label
    class="field__label field__label--end"
    for="radio-30"
  >
    Custom style
  </label>
</DocumentFragment>
`;

exports[`Storyshots ebay-radio Using custom label html 1`] = `
<DocumentFragment>
  <p>
    <span
      class="radio"
      style="align-items: center;"
    >
      <input
        class="radio__control"
        id="radio-1"
        type="radio"
        value="123"
      />
      <span
        class="radio__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="radio__checked icon icon--radio-checked"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-radio-checked"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="radio__unchecked icon icon--radio-unchecked"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-radio-unchecked"
          />
        </svg>
      </span>
    </span>
    <label
      class="field__label field__label--end"
      for="radio-1"
    >
      Default
    </label>
  </p>
  <p>
    <span
      class="radio radio--large"
      style="align-items: center;"
    >
      <input
        class="radio__control"
        id="radio-11"
        type="radio"
        value="123"
      />
      <span
        class="radio__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="radio__checked icon icon--radio-checked-large"
          focusable="false"
          height="64px"
          width="64px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-radio-checked-large"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="radio__unchecked icon icon--radio-unchecked-large"
          focusable="false"
          height="64px"
          width="64px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-radio-unchecked-large"
          />
        </svg>
      </span>
    </span>
    <label
      class="field__label field__label--end"
      for="radio-11"
    >
      Large
    </label>
  </p>
</DocumentFragment>
`;
